packages/pwa/app/commerce-api/utils.js

```diff
@@ -285,6 +246,21 @@ export const isError = (jsonResponse) => {
     return false
 }
 
+/**
+ * Decorator that wraps async functions to handle error response.
+ * @param {function} func - The async function that gets wrapped
+ * @returns {function}
+ */
+export const handleAsyncError = (func) => {
```

Comment:
The blur for this utility says that the function passed in should be `async`. It's probably worth testing the function and error'ing out if it's not.

Something like this:

```
if (func.constructor.name !== `AsyncFunction`) {
    throw new Error('Parameter func must be asynchronous')
}
```

Reply:
sorry my comment is misleading and I corrected it.

As long as the `func` is a function and it returns a promise, it doesn't need to use the syntactic suger `async`.

Reply:
i'll add some tests